src/main/java/org/broadinstitute/hellbender/tools/variantdb/arrays/ArrayExtractCohortEngine.java

```diff
@@ -266,9 +274,108 @@ private VariantContext createVariantContextFromSampleRecord(final ProbeInfo prob
+                genotypeAlleles.add(alleles.get(2));
+                break;
+            case MISSING:
+            default:
+                genotypeAlleles.add(Allele.NO_CALL);
+                genotypeAlleles.add(Allele.NO_CALL);
+                break;
+        }
+
+        genotypeBuilder.alleles(genotypeAlleles);
+
+        if (!gtDataOnly) {
+            genotypeBuilder.attribute(RawArrayTsvCreator.NORMX, formatFloatForVcf(getNullableFloatFromDouble(sampleRecord.get(RawArrayFieldEnum.NORMX.name()))));
+            genotypeBuilder.attribute(RawArrayTsvCreator.NORMY, formatFloatForVcf(getNullableFloatFromDouble(sampleRecord.get(RawArrayFieldEnum.NORMY.name()))));
+            genotypeBuilder.attribute(RawArrayTsvCreator.BAF, formatFloatForVcf(getNullableFloatFromDouble(sampleRecord.get(RawArrayFieldEnum.BAF.name()))));
+            genotypeBuilder.attribute(RawArrayTsvCreator.LRR, formatFloatForVcf(getNullableFloatFromDouble(sampleRecord.get(RawArrayFieldEnum.LRR.name()))));
+        }
+
+        genotypeBuilder.name(sample);
+
+        builder.genotypes(genotypeBuilder.make());
+
+        try {
+            VariantContext vc = builder.make();
+            return vc;
+        } catch (Exception e) {
+            System.out.println("Error: "+ e.getMessage() + " processing " + sampleRecord + " PI: " + probeInfo.alleleA + "/" +probeInfo.alleleB + " with ga " + genotypeAlleles + " and alleles " + alleles);
+            throw e;
+        }
+
+    }
+
+    List<Allele> createAllelesFromProbeInfo(final ProbeInfo probeInfo) {
         final List<Allele> alleles = new ArrayList<>();
-        Allele ref = Allele.create(probeInfo.ref, true);        
+        Allele ref = Allele.create(probeInfo.ref, true);
+        alleles.add(ref);
+
+        Allele alleleA = Allele.create(probeInfo.alleleA, false);
+        Allele alleleB = Allele.create(probeInfo.alleleB, false);
+
+        boolean alleleAisRef = probeInfo.ref.equals(probeInfo.alleleA);
+        boolean alleleBisRef = probeInfo.ref.equals(probeInfo.alleleB);
+
+        if (alleleAisRef) {
+            alleleA = ref;
+        } else {
+            alleles.add(alleleA);
+        }
+
+        if (alleleBisRef) {
+            alleleB = ref;
+        } else {
+            alleles.add(alleleB);
+        }
+        return alleles;
+    }
+
+        private VariantContext createVariantContextFromSampleRecordLegacyGT(final ProbeInfo probeInfo, final GenericRecord sampleRecord, final Set<String> columnNames, final String contig, final long startPosition, final String sample) {
+        final VariantContextBuilder builder = new VariantContextBuilder();
+        final GenotypeBuilder genotypeBuilder = new GenotypeBuilder();
+
+        builder.chr(contig);
+        builder.start(startPosition);
+        builder.id(probeInfo.name);
+
+        List<Allele> alleles = createAllelesFromProbeInfo(probeInfo);
+        Allele ref = Allele.create(probeInfo.ref, true);
```

Comment:
Is the diff just being weird or should the code below be deleted since it was moved to `createAllelesFromProbeInfo`?

Reply:
oh, i need to change createAlleles... back to just new ArrayList.
this version of the methods need to know alleleA and alleleB so there is some duplicate code while we are supporting the legacy mode.